src/app/qgisapp.cpp

```diff
@@ -7460,7 +7460,7 @@ bool QgisApp::openLayer( const QString &fileName, bool allowInteractive )
 
   // highest priority = delegate to provider registry to handle
   const QList< QgsProviderRegistry::ProviderCandidateDetails > candidateProviders = QgsProviderRegistry::instance()->preferredProvidersForUri( fileName );
-  if ( candidateProviders.size() == 1 && candidateProviders.at( 0 ).layerTypes().size() == 1 )
+  if ( candidateProviders.size() >= 1 && candidateProviders.at( 0 ).layerTypes().size() == 1 )
   {
```

Comment:
This is not right -- that branch only applies when there's a single candidate

Reply:
Actually .copc.laz file could be read by using both pdal (as a normal laz file) or with the COPC provider. 
So preferredProvidersForUri is returning both providers...

Reply:
You could add .copc.laz to the Blocklist of URIs for the pdal provider.

Reply:
You're right, I put a higher priority for the copc provider (101) and now preferredProvidersForUri returns just one provider.